src/completion.ts

```diff
@@ -214,6 +215,8 @@ class DictionaryCompletionItemProvider implements vscode.CompletionItemProvider
         }
 
         switch (this.fileType) {
+            case "asciidoc":
+                return this.completeByFirstLetter(firstLetter, addSpace);
```

Comment:
Looks good. But I guess there are some cases where we do not want a completion (e.g. urls). You can use [regex101](https://regex101.com/) (select ECMAScript in the left panel) to help you write those rules.